cni/cmd/tc-redirect-tap/main.go

```diff
@@ -135,7 +137,21 @@ func newPlugin(args *skel.CmdArgs) (*plugin, error) {
 		netNS: netNS,
 
 		currentResult: currentResult,
-	}, nil
+	}
+
+	tapName := extractArgs(args.Args, TC_REDIRECT_TAP_NAME)
+	if tapName != "" {
+		plugin.tapName = tapName
+	}
+	tapUID, _ := strconv.Atoi(extractArgs(args.Args, TC_REDIRECT_TAP_UID))
```

Comment:
Actually, in current implementation, it's not possible without any intervention in behavior, since those params are not mandatory, we cannot  return errors in case they just wasn't provided, but this will happen coz by default extractArgs returns "" and this will not be properly converted in int with atoi function.

Reply:
Right, so we need to distinguish between
1. the arg being provided and valid
2. the arg being provided but invalid
3. the arg not being provided at all

Maybe a good way to implement this would be to generalize the approach and have `extractArgs` just convert the whole `args` string into a `map[string]string` of the key/val pairs and return that map. Then we can use the map's builtin ability to test whether a key is present, something like:
```
parsedArgs := extractArgs(args.Args)

var err error
var tapUID int
if tapUIDVal, wasDefined := parsedArgs[TCRedirectTapName]; wasDefined {
    tapUID, err = strconv.Atoi(tapUIDVal)
    if err != nil {
       // return error saying that tapUID was an invalid format 
    }
}
```